### PR TITLE
magento/28577_undeclared_webapi_dependency-fix

### LINKED ISSUE #28577: GraphQL module has undeclared dependency on Webapi module
GraphQL module relies on Webapi module for authentication and if Webapi module is disabled - some GraphQL queries will become inaccessible.

Ideally, dependency should be fixed or at least declared as 'required' instead of 'suggest'.

Severity is not critical because it is unlikely that someone will disable Webapi module, but in the future it should be possible to set up GraphQL-only instance.

 !image-2019-10-09-15-41-21-970.png!

### PR DESCRIPTION
**Pull request by sasha19957099**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. ...
2. ...

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

File: composer.lock [hunk 1]
```diff
@@ -210,24 +210,24 @@
         },
         {
             "name": "composer/composer",
-            "version": "1.10.6",
+            "version": "1.10.7",
```
Comment (danielrenaud) [inline composer.lock]: We don't want to update dependencies as a part of this change.
You can use `composer update -lock` to only update composer.lock hash without updating dependencies

**Review by danielrenaud — CHANGES_REQUESTED** [review]

**Comment by sasha19957099** [comment]
@magento run all tests

**Review by danielrenaud — APPROVED** [review]

**Comment by sasha19957099** [comment]
@magento run all tests

**Comment by sasha19957099** [comment]
@magento run all tests

**Comment by sasha19957099** [comment]
@magento run all tests

**Comment by danielrenaud** [comment]
Manual testing not needed since this doesn't touch functionality. Automated tests will be enough

**Review by danielrenaud — APPROVED** [review]

File: dev/tests/static/testsuite/Magento/Test/Integrity/_files/dependency_test/whitelist/redundant_dependencies_webapi.php [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+return [
+    'Magento\GraphQl' => [
+        'Magento\Webapi' => 'Magento\Webapi'
+    ]
+];
```
Comment (ihor-sviziev) [inline dev/tests/static/testsuite/Magento/Test/Integrity/_files/dependency_test/whitelist/redundant_dependencies_webapi.php:12]: Hi @sasha19957099 @danielrenaud,
Could you explain why this file was added? Seems like test shouldn't fail anymore as it has declared dependency on web api module. An I wrong?

**Pull request merged**